### PR TITLE
Read/write relative paths to gpkg rasters in project (when enabled)

### PR DESCRIPTION
**Pull request by wonder-sk**
Routines in raster layer were not aware of "GDAL:filename:table" syntax used for raster layers from GeoPackages so project files were storing absolute paths even when relative paths should have been used.

Parsing of file name done just like GDAL does in GDALGeoPackageDataset::Open()
https://github.com/OSGeo/gdal/blob/master/gdal/ogr/ogrsf_frmts/gpkg/ogrgeopackagedatasource.cpp#L783

### CODE REVIEW

File: src/core/raster/qgsrasterlayer.cpp [hunk 1]
```diff
@@ -1646,6 +1646,32 @@ bool QgsRasterLayer::writeXml( QDomNode &layer_node,
   return writeSymbology( layer_node, document, errorMsg, context );
 }
 
+// TODO: this should ideally go to gdal provider (together with most of encodedSource() + decodedSource())
```
Comment (nyalldawson) [inline src/core/raster/qgsrasterlayer.cpp]: Hopefully not being too annoying... but what's stopping this? Do we need to make a writeSource/readSource method available to provider libraries?

Comment (nyalldawson) [reply src/core/raster/qgsrasterlayer.cpp]: ok, proceed. ;)

**Comment by agiudiceandrea** [comment]
Hi @wonder-sk, will this patch also fix this issue https://issues.qgis.org/issues/20050 about "GPKG and SQLite files containing raster layers are incorrectly added to the map"?

**Comment by wonder-sk** [comment]
@agiudiceandrea not really - this PR is really just about making sure that we are able to convert paths to relative so that pass around projects with geopackages without getting projects broken.

**Pull request merged**